### PR TITLE
Add utilities for checking code

### PR DESCRIPTION
**Pull request by rbberger**
**Summary**

* Adds whitespace utility to check and fix common whitespace and file encoding issues
* Adds `check-whitespace` and `fix-whitespace` targets to cmake build
* Adds check for file permissions (e.g. source files or input scripts shouldn't be executable)
* Add `check-permissions` and `fix-permissions` targets to cmake build

**Author(s)**

@rbberger 

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: cmake/Modules/CodingStandard.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+if(CMAKE_VERSION VERSION_LESS 3.12)
+    find_package(PythonInterp) # Deprecated since version 3.12
```
Comment (akohlmey) [inline cmake/Modules/CodingStandard.cmake:2]: This should be looking for Python 3. So something more similar to https://github.com/lammps/lammps/blob/ec6d3586ce30a4fd62562272b264c6ad9453c567/cmake/Modules/Documentation.cmake

**Comment by akohlmey** [comment]
@rbberger perhaps it would make sense to also include clang-format processing (#2157) into this and make it a make target, too. But there would need to be some way to apply it only to selected folders (either through some way that is internal to clang-format or some (per folder?) signaling that the make command will check for).

**Comment by junghans** [comment]
> @rbberger perhaps it would make sense to also include clang-format processing (#2157) into this and make it a make target, too. But there would need to be some way to apply it only to selected folders (either through some way that is internal to clang-format or some (per folder?) signaling that the make command will check for).

@rbberger have a look here:
https://github.com/votca/votca/blob/d022cdf131d32a315dd7f52d09869e022d2f3852/share/format/CMakeLists.txt#L1-L12
with
https://github.com/votca/votca/blob/d022cdf131d32a315dd7f52d09869e022d2f3852/CMakeModules/FindCLANG_FORMAT.cmake#L22-L51

One caveat with clang-format is that different major versions will format the code slightly differently.

**Comment by rbberger** [comment]
See https://github.com/lammps/lammps/pull/2157/commits/e840fa23f11cca188c5e0ddc27ce8155ea2e4299

**Pull request merged**